android/FirebaseContinue/library/src/main/java/com/firebasecontinue/FirebaseContinue.java

```diff
@@ -0,0 +1,174 @@
+                // before setting a value any existing value must first be deleted.
+
+                final TaskCompletionSource<DatabaseReference> deletionTaskCompletion =
+                        new TaskCompletionSource<>();
+                DatabaseReference mostRecentActivityRef = task.getResult();
+                mostRecentActivityRef.removeValue(
+                        new DatabaseReference.CompletionListener() {
+                            @Override
+                            public void onComplete(DatabaseError error, DatabaseReference ref) {
+                                if (error == null) {
+                                    // Set that this Task was successful and pass along the
+                                    // DatabaseReference to the next Task.
+                                    deletionTaskCompletion.setResult(ref);
+                                } else {
+                                    // Set that this Task was unsuccessful.
+                                    deletionTaskCompletion.setException(error.toException());
+                                }
+                            }
+                        });
+
+                return deletionTaskCompletion.getTask();
+            }
+        }).continueWithTask(new Continuation<DatabaseReference, Task<Void>>() {
+            @Override
+            public Task<Void> then(Task<DatabaseReference> task) throws Exception {
+                // Now we can set the new value for the most recent Activity the user may wish
+                // to continue within the application, since any previous value has been deleted.
+
+                // Create a Firebase Continue Activity to use as the new most recent Activity
+                // the user may wish to continue within the application.
+                // The schema of each Activity is defined in
+                // sample-firebase-continue-database.rules.json.
+                Map<String, Object> activityMetadata = new HashMap<>();
+                activityMetadata.put("addedAt", ServerValue.TIMESTAMP);
+                Map<String, Object> activity = new HashMap<>();
+                activity.put("url", activityUrl);
+                activity.put("metadata", activityMetadata);
+
+                final TaskCompletionSource<Void> setValueTaskCompletion =
+                        new TaskCompletionSource<>();
+                DatabaseReference mostRecentActivityRef = task.getResult();
+                mostRecentActivityRef.setValue(
+                        activity,
+                        new DatabaseReference.CompletionListener() {
+                            @Override
+                            public void onComplete(DatabaseError error, DatabaseReference ref) {
+                                if (error == null) {
+                                    // Set that this Task was successful.
+                                    setValueTaskCompletion.setResult(null);
```

Comment:
I don't think so because that would give the caller in the application which uses this library the DatabaseReference (in a SuccessListener), which I do not believe they should have a need for.

What do you think?